apps/web/src/components/TablePaginationFooter/GoToPageInput/index.tsx

```diff
@@ -24,15 +24,18 @@ export function GoToPageInput({
     event.preventDefault()
 
     const targetPage = Number(new FormData(event.currentTarget).get('page'))
-    const queryParams = window.location.search
+    const urlParams = new URLSearchParams(window.location.search)
+
+    // Replace or add the page parameter
+    urlParams.set('page', targetPage.toString())
 
     if (baseUrl) {
       router.push(
         buildPaginatedUrl({
           baseUrl,
           page: targetPage,
           pageSize,
-          queryParams,
+          queryParams: urlParams.toString(),
           encodeQueryParams: false,
```

Comment:
this was broken it was adding 2 page search params instead of replacing the existing one

Reply:
This is used also in logs. Can you check it works ok when logs has filters and go to page 2 or 3...?